src/Solnet.Rpc/Core/Http/JsonRpcClient.cs

```diff
@@ -66,6 +74,10 @@ protected async Task<RequestResult<T>> SendRequest<T>(JsonRpcRequest req)
 
             try
             {
+                // pre-flight check with rate limiter
+                _rateLimiter.Fire();

```

Comment:
Instead of instancing a default rate limiter for non-rate limited connections, I'd rather we skip it completely. `_rateLimiter?.Fire();`